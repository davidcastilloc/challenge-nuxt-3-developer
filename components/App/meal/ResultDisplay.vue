<template>
  <BasePlaceholderPage :title="`You could make ${meals} lunches with this recipe`">
    <template #image>
      <img src="/images/display-recipe.jpg" alt="Placeholder" class="w-full rounded-md mt-4" />
      <p class="text-2xl font-bold">
        The secret formula is
      </p>
      <div class="capitalize flex" v-for="(ingredient, key) in ingredients" :key="key">
        <img :src="`/icons/${[key]}.gif`" class="relative h-6 w-6" />
        <span class="ml-2 capitalize">{{ key }}: {{
          ingredient
        }} </span>
      </div>
    </template>
    <div class="mt-2 flex justify-center gap-2">
      <BaseButton to="/" color="primary" class="w-full mb-2">
        Back to Calculator
      </BaseButton>
    </div>
  </BasePlaceholderPage>
</template>

<script lang="ts" setup>
defineProps({
  meals: {
    type: String,
    required: true,
  },
  ingredients: {
    type: Object,
    required: true,
  },
})

</script>

<style></style>