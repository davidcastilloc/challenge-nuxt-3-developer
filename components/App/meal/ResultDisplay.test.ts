// define test for ResultDisplay component
import { it, expect, test } from "vitest";
import { mountSuspended } from "@nuxt/test-utils/runtime";
import ResultDisplay from "./ResultDisplay.vue";

test("ResultDisplay component renders correctly", async () => {
  const wrapper = mountSuspended(ResultDisplay, {
    props: {
      meals: "2",
      ingredients: {
        pasta: 500,
        bacon: 200,
        eggs: 1,
        milk: 200,
        butter: 500,
        oil: 100,
      },
    },
  });

  expect((await wrapper).html()).toMatchSnapshot();
});
