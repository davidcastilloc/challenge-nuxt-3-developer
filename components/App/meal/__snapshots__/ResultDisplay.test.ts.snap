// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ResultDisplay component renders correctly 1`] = `
"<div class="nui-placeholder-page nui-placeholder-xs">
  <div class="nui-placeholder-page-inner">
    <div class="nui-placeholder-page-img"><img src="/images/display-recipe.jpg" alt="Placeholder" class="w-full rounded-md mt-4">
      <p class="text-2xl font-bold"> The secret formula is </p>
      <div class="capitalize flex"><img src="/icons/pasta.gif" class="relative h-6 w-6"><span class="ml-2 capitalize">pasta: 500</span></div>
      <div class="capitalize flex"><img src="/icons/bacon.gif" class="relative h-6 w-6"><span class="ml-2 capitalize">bacon: 200</span></div>
      <div class="capitalize flex"><img src="/icons/eggs.gif" class="relative h-6 w-6"><span class="ml-2 capitalize">eggs: 1</span></div>
      <div class="capitalize flex"><img src="/icons/milk.gif" class="relative h-6 w-6"><span class="ml-2 capitalize">milk: 200</span></div>
      <div class="capitalize flex"><img src="/icons/butter.gif" class="relative h-6 w-6"><span class="ml-2 capitalize">butter: 500</span></div>
      <div class="capitalize flex"><img src="/icons/oil.gif" class="relative h-6 w-6"><span class="ml-2 capitalize">oil: 100</span></div>
    </div>
    <div class="nui-placeholder-page-content">
      <h4 class="nui-heading nui-heading-xl nui-weight-medium nui-lead-normal nui-placeholder-page-title">You could make 2 lunches with this recipe</h4>
      <!--v-if-->
      <div class="mt-2 flex justify-center gap-2"><a href="/" disabled="false" class="nui-button nui-button-md nui-button-rounded-sm nui-button-solid nui-button-primary w-full mb-2"> Back to Calculator
          <!--v-if-->
        </a></div>
    </div>
  </div>
</div>"
`;
